### PR TITLE
Fix monotonicity check for DateTime64

### PR DESCRIPTION
**Pull request by antonio2368**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in official stable or prestable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Correctly check for the monotonicity of a range when `DateTime64` column is used.

Caught by fuzzer:
https://s3.amazonaws.com/clickhouse-test-reports/0/956ef6bd2133ce93524734a1158483c4ecb185c9/fuzzer_astfuzzerdebug/report.html

I'm not 100% confident in my understanding of DateTime related code so maybe it's not the best solution.
### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: tests/queries/0_stateless/02680_datetime64_monotonic_check.sql [hunk 1]
```diff
@@ -0,0 +1,15 @@
+DROP TABLE IF EXISTS 02680_datetime64_monotonic_check;
```
Comment (hanfei1991) [inline tests/queries/0_stateless/02680_datetime64_monotonic_check.sql:1]: I run this test on master and got same result

Comment (antonio2368) [reply tests/queries/0_stateless/02680_datetime64_monotonic_check.sql:1]: did you run with the debug or release build?
the LOGIC_ERROR will cause abort in debug build.

Comment (hanfei1991) [reply tests/queries/0_stateless/02680_datetime64_monotonic_check.sql:1]: Well I ran it in the release build ...

**Review by hanfei1991 — APPROVED** [review]
LGTM

**Comment by tavplubix** [comment]
The new assertion `checkAndGetDataType<DataTypeDateTime64>(type_ptr)` has failed: https://s3.amazonaws.com/clickhouse-test-reports/0/4303b91ee71d82281f221d85e47e593ffb2ecd95/fuzzer_astfuzzerdebug/report.html

**Comment by antonio2368** [comment]
Fixed here https://github.com/ClickHouse/ClickHouse/pull/47860

**Pull request merged**